src/Symfony/Component/Console/Style/StyleInterface.php

```diff
@@ -87,65 +83,53 @@ public function table(array $headers, array $rows);
     /**
      * Asks a question.
      *
-     * @param string        $question
-     * @param string|null   $default
      * @param callable|null $validator
      *
      * @return mixed
      */
-    public function ask($question, $default = null, $validator = null);
+    public function ask(string $question, ?string $default = null, $validator = null);
```

Comment:
can't we add the callable type hint for `$validator`?
extra `?` for `$default` also, there is a bunch of them in this PR

Reply:
We were talking about callable being dangerous to type hint because of the perf that could be touched by that.

Reply:
Any link? :)

Reply:
> Adding a callable type-hint triggers autoloading if the passed value contains a class reference as string. This is a side-effect that should be considered.
in https://github.com/symfony/symfony/issues/32179

Reply:
if you think that does not change anything we can do it :)

Reply:
if you think that does not change anything we can do it :)

Reply:
I think, the `callable` type hint is safe here. The callback is meant to be called right away, so we don‘t have to worry about unnecessary autoloading.